### ISSUE TITLE
CSS variables not supported in GMail - AMP

### ISSUE THREAD

**Issue by kieltyka** [ampproject/amp.dev#3274]
### Current Behavior
AMP 4 email "Star rating" works appropriately in the amp [playground ](https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Finteractivity-dynamic-content%2Fstar_rating&format=websites&_gl=1*f7bn79*_ga*YW1wLWNxck55bm9aLTZRSlkzTmdUMHZZNXc.
)

With the stars rendering as gold when selected / submitted

However when actually sending an email to a Gmail inbox, the stars are not filled with gold, but instead filled with white (as are the outlines) resulting in the design looking off and making it not usable

Please also provide:
- Page link: <insert the URL link to the amp.dev page(s) that you found the issue on>
- (Optional) Screenshots / videos

### Expected behavior/code
Should be able to see the gold stars render in the email content as we see it in the AMP 4 Email and AMPHTML test pages

I've provided my code below, but removed the image and action-xhr endpoint

```
<!-- -->
<!doctype html>
<html ⚡4email>
<head>
<meta charset="utf-8">
<script async src="https://cdn.ampproject.org/v0.js"></script>
<script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
<style amp4email-boilerplate>body{visibility:hidden}</style>
<style amp-custom>
.container {
max-width: 500px;
margin: auto;
font-family: sans-serif;
padding: 1em;
text-align: center;
}

.block {
display: block;
width: 100%;
}

.m1 {
margin: 1em 0;
}

label {
margin-bottom: 0.5em;
}
.rating {
--star-size: 3; /* use CSS variables to calculate dependent dimensions later */
padding: 0; /* to prevent flicker when mousing over padding */
border: none; /* to prevent flicker when mousing over border */
unicode-bidi: bidi-override; direction: rtl; /* for CSS-only style change on hover */
text-align: center; /* revert the RTL direction */
user-select: none; /* disable mouse/touch selection */
font-size: 3em; /* fallback - IE doesn't support CSS variables */
font-size: calc(var(--star-size) * 1em); /* because `var(--star-size)em` would be too good to be true */
cursor: pointer;
-webkit-tap-highlight-color: rgba(0,0,0,0);
-webkit-tap-highlight-color: transparent;
}
/* the stars */
.rating > label {
display: inline-block;
position: relative;
width: 1.1em; /* magic number to overlap the radio buttons on top of the stars */
width: calc(var(--star-size) / 3 * 1.1em);
}
.rating > *:hover,
.rating > *:hover ~ label,
.rating:not(:hover) > input:checked ~ label {
color: transparent; /* reveal the contour/white star from the HTML markup */
cursor: inherit; /* avoid a cursor transition from arrow/pointer to text selection */
}
.rating > *:hover:before,
.rating > *:hover ~ label:before,
.rating:not(:hover) > input:checked ~ label:before {
content: "★";
position: absolute;
left: 0;
color: gold;
}
.rating > input {
position: relative;
transform: scale(3); /* make the radio buttons big; they don't inherit font-size */
transform: scale(var(--star-size));
/* the magic numbers below correlate with the font-size */
top: -0.5em; /* margin-top doesn't work */
top: calc(var(--star-size) / 6 * -1em);
margin-left: -2.5em; /* overlap the radio buttons exactly under the stars */
margin-left: calc(var(--star-size) / 6 * -5em);
z-index: 2; /* bring the button above the stars so it captures touches/clicks */
opacity: 0; /* comment to see where the radio buttons are */
font-size: initial; /* reset to default */
}
form.amp-form-submit-error [submit-error] {
color: red;
}

form.amp-form-submit-success > input {
display: none
}
</style>
</head>
<body>
<div class="container">
<!--
The main content of the email, an image and a short description of the event.
-->
<div>
<amp-img class="m1" width="600" height="600" layout="responsive" src="img.com"></amp-img>
<p>We hope you enjoyed using --- for your recent trip!</p>
</div>
<hr>
<p>How would you rate your experience?</p>
<form id="rating" method="post" action-xhr="XXXX">
<fieldset class="rating">

<input name="rating" type="radio" id="rating5" value="5" on="change:step2.show">
  <label for="rating5" title="5 stars">☆</label>

  <input name="rating" type="radio" id="rating4" value="4" on="change:step2.show">
  <label for="rating4" title="4 stars">☆</label>

  <input name="rating" type="radio" id="rating3" value="3" on="change:step2.show">
  <label for="rating3" title="3 stars">☆</label>

  <input name="rating" type="radio" id="rating2" value="2" on="change:step2.show">
  <label for="rating2" title="2 stars">☆</label>

  <input name="rating" type="radio" id="rating1" value="1" on="change:step2.show">
  <label for="rating1" title="1 stars">☆</label>
</fieldset>

<div class="m1" id="step2" hidden>
<label class="block" for="info">Would you like to tell us more?</label>
<textarea class="block" id="info" name="feedback" rows="5"></textarea>
</div>
<input type="submit" value="Send feedback">
<input type="reset" value="Clear">

<div class="m1" submit-success>
Thank you for submitting feedback.
</div>
```
### How do we reproduce the issue?

Please provide the steps to reproduce the issue:

1. Send email to gmail inbox
2. Select a star rating.

What browser are you using? Chrome
What O/S are you using? Mac OS Catalina

**Comment by M-J-Robbins** [comment]
A quick look at that code and there is quite a bit not supported by Gmail in there

Interestingly Gmail actually seems to recognise the reading of CSS variables `var(--star-size)` but not the setting of them `--star-size: 3;` 

From that code sample it appears that Gmail AMP doesn't support,
* `user-select:`
* `-webkit-tap-highlight-color`
* `cursor`
* `:not()`
* `:before`
* attribute selectors `[submit-error]`
* negative margin `margin-left: -2.5em;`

**Comment by sebastianbenz** [comment]
//cc @fstanis

**Comment by fstanis** [comment]
Regarding things not supported or not working in Gmail: if they are listed in [CSS spec for AMP for Email](https://amp.dev/documentation/guides-and-tutorials/learn/email-spec/amp-email-css/?format=email), such as `:not()` which should be supported, then this is likely a bug in Gmail. Please [file bugs in Gmail related to AMP here](https://issuetracker.google.com/issues/new?component=592249&template=1238934) with information on how to reproduce. If it's not in the spec, we need a spec change, which should probably be [discussed in ampproject/wg-amp4email](https://github.com/ampproject/wg-amp4email/issues).

Regarding this sample, there are two options here:

- Remove the non-essential things unsupported in emails (e.g. variables) and leave those that aren't as impactful (e.g. `-webkit-tap-highlight-color`).
- Fork the sample and have an email version of it.

@sebastianbenz let me know how you propose we proceed.

**Comment by patrickkettner** [comment]
this was updated and fixed, via #5280
